runtime-modules/content/src/lib.rs

```diff
@@ -1259,8 +1307,72 @@ decl_module! {
         ) {
             Self::not_implemented()?;
         }
+
+        #[weight = 10_000_000] // TODO: adjust weight
+        pub fn update_commitment(
+            origin,
+            new_commitment: <T as frame_system::Trait>::Hash,
+        ) {
+        ensure_root(origin)?;
+            let old_commitment = <Commitment<T>>::get();
+            if old_commitment == new_commitment { return Ok(()) }
+            <Commitment<T>>::put(new_commitment);
+            Self::deposit_event(RawEvent::CommitmentUpdated(new_commitment));
+        }
+
+    #[weight = 10_000_000]
+    pub fn update_channel_reward(
+            origin,
+            proof: PullPaymentProof<T>,
+    ) {
+
+        let signing_acc = ensure_signed(origin)?;
+        let elem = &proof.data;
+
+        let mut channel = ChannelById::<T>::get(elem.channel_id);
```

Comment:
This will not work, this will always return a channel, even if there if there is no pair in the map, because it returns a default constructed channel. 

1. introduce a test that fails as a result of this bug
2. check how channel checking is done in other extrinsics, e.g. `update_channel`.
3. confirm new test now passes